DependencyInjection/Compiler/RepositoryPass.php

```diff
@@ -29,7 +29,21 @@ public function process(ContainerBuilder $container)
     {
         $managers = $container->getParameter('es.managers');
 
+        $hasContainerBuildId = $container->hasParameter('container.build_id');
+        if (!$hasContainerBuildId) {
+            // the 'container.build_id' is required for `es.cache_engine` system cache which normally can not
+            // be constructor inside a compiler pass. This is a workaround to make it work.
+            // see also:
+            //     - https://github.com/symfony/symfony/blob/52a92926f7fed15cdff399c6921100a10e0d6f61/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php#L389
+            //     - https://github.com/symfony/symfony/blob/52a92926f7fed15cdff399c6921100a10e0d6f61/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L2322
+            $hasContainerBuildId = true;
+            $container->setParameter('container.build_id', hash('crc32', 'Abc123' . time()));
+        }
+
         $collector = $container->get('es.metadata_collector');
+        if (!$hasContainerBuildId) {
+            $container->getParameterBag()->remove('container.build_id');
+        }
```

Comment:
This is required when using the `cache.system` and access it inside the compile time. Another possibility would be to use the `cache.app` but which does not require build_id. But metadata normally is invalidated between deployments and should fast readable which normally is job of the `cache.system` and not the `cache.app`.